torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -709,15 +712,39 @@ void SegmentedFusion::finalize() {
 
   // Go through all edges of the segmented fusion.
   for (auto edge : edges()) {
+
+    TORCH_INTERNAL_ASSERT(edge->val->isA<TensorView>());
```

Comment:
Just curious if a case where a scalar edge was encountered. I don't think we currently create any edges containing scalars since our kernels cannot pass scalar outputs.

Reply:
No, I just don't like assuming something we don't assert as code can change and mistakes can be made.

Reply:
I see, makes sense. Thanks!